### PR TITLE
add cosign to ci

### PR DESCRIPTION
**Pull request by JasonPowr**

### CODE REVIEW

File: ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+binary_build_commands: make cosign
+build_root:
+  image_stream_tag:
+    name: release
```
Comment (danilo-gemoli) [inline ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml:4]: We don't have `golang-1.20` yet:
```sh
$ oc --context=app.ci -n openshift get istag/release:golang-1.20
Error from server (NotFound): imagestreamtags.image.openshift.io "release:golang-1.20" not found
```

Comment (lance) [reply ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml]: @JasonPowr I believe you should be able to build with Go 1.20 using this image stream tag.

https://github.com/openshift/release/blob/1bd82837937cf107e83a32900496f3c61b5ccc2f/ci-operator/config/securesign/rekor/securesign-rekor-redhat-v1.2.2.yaml#L2-L5

File: ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: unit
+  commands: mkdir -p /.sigstore && chmod -R 777 /.sigstore && GOFLAGS="" make test
```
Comment (lance) [inline ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml:26]: What is the purpose of the `mkdir` and `chmod` commands here? If they are needed, shouldn't the be in the local dir, e.g. `./.sigstore`?

File: ci-operator/config/securesign/cosign/OWNERS [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# The OWNERS file is used by prow to automatically merge approved PRs.
+
+approvers:
+- lance
+- sabre1041
+- cooktheryan
+- sallyom
```
Comment (lance) [inline ci-operator/config/securesign/cosign/OWNERS:7]: @JasonPowr if you add yourself and @tommyd450 to this file, you both will have /lgtm and /approve rights on this repo.

File: ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: unit
+  commands: |
+    export HOME="/tmp/home"
+    mkdir -p "${HOME}"
```
Comment (lance) [inline ci-operator/config/securesign/cosign/securesign-cosign-release-next.yaml:28]: 👍

File: ci-operator/jobs/securesign/cosign/OWNERS [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# The OWNERS file is used by prow to automatically merge approved PRs.
+
+approvers:
+- lance
+- sabre1041
+- cooktheryan
+- sallyom
```
Comment (lance) [inline ci-operator/jobs/securesign/cosign/OWNERS:7]: As above - add yourself and @tommyd450 here

File: core-services/prow/02_config/securesign/cosign/OWNERS [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# The OWNERS file is used by prow to automatically merge approved PRs.
+
+approvers:
+- lance
+- sabre1041
+- cooktheryan
+- sallyom
```
Comment (lance) [inline core-services/prow/02_config/securesign/cosign/OWNERS:7]: As above - add you and @tommyd450 here.

**Review by lance — COMMENTED** [review]
Overall looks good to me. You may want to go ahead and add yourself and Tommy as I suggested.

**Pull request merged**